.github/workflows/cypress-update-snapshots.yml

```diff
@@ -0,0 +1,79 @@
+name: Cypress update snapshots
+
+on:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      testname:
+        description: "Enter the test name to run (see e2e/specs/[testname].spec.js). For example: st_alert"
+        default: "st_alert"
+        type: "string"
+
+# Avoid duplicate workflows on same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-cypress-update-snapshots
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash --login -eo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+            python-version: "3.10"
```

Comment:
We updated our Python setup to source the python major version from a custom action - [Example here](https://github.com/streamlit/streamlit/blob/develop/.github/workflows/js-tests.yml#L37-L42).
```      
- name: Set Python version vars
   uses: ./.github/actions/set_python_versions
- name: Set up Python ${{ env.PYTHON_MAX_VERSION }}
   uses: actions/setup-python@v4
   with:
      python-version: "${{ env.PYTHON_MAX_VERSION }}"
```